src/components/RichSelect/__stories__/index.stories.tsx

```diff
@@ -0,0 +1,346 @@
+        <FlexBox alignItems="center">
+          Option B
+          <Badge size="rounded" ml={1}>
+            Awesome badge
+          </Badge>
+        </FlexBox>
+      </RichSelect.Option>
+    </RichSelect>
+  ),
+]
+
+export const NoLabel = Template.bind({})
+NoLabel.parameters = {
+  docs: {
+    storyDescription: 'Use `noTopLabel` to hide the label.',
+  },
+}
+NoLabel.decorators = [
+  () => (
+    <RichSelect name="label" noTopLabel>
+      <RichSelect.Option value="a">Option A</RichSelect.Option>
+      <RichSelect.Option value="b">Option B</RichSelect.Option>
+    </RichSelect>
+  ),
+]
+
+export const Searchable = Template.bind({})
+Searchable.parameters = {
+  docs: {
+    storyDescription: 'Use `isSearchable` prop to make option searchable.',
+  },
+}
+Searchable.decorators = [
+  () => (
+    <RichSelect name="searchable" isSearchable>
+      <RichSelect.Option value="a">Option A</RichSelect.Option>
+      <RichSelect.Option value="b">Option B</RichSelect.Option>
+    </RichSelect>
+  ),
+]
+
+export const Required = Template.bind({})
+Required.parameters = {
+  docs: {
+    storyDescription: 'Use `required` to make this input mandatory',
+  },
+}
+Required.decorators = [
+  () => (
+    <RichSelect name="required" required>
+      <RichSelect.Option value="a">Option A</RichSelect.Option>
+      <RichSelect.Option value="b">Option B</RichSelect.Option>
+    </RichSelect>
+  ),
+]
+
+export const Disabled = Template.bind({})
+Disabled.parameters = {
+  docs: {
+    storyDescription: 'Use `disabled` to disable this input',
+  },
+}
+Disabled.decorators = [
+  () => (
+    <RichSelect name="disabled" disabled value="a">
+      <RichSelect.Option value="a">Option A</RichSelect.Option>
+      <RichSelect.Option value="b">Option B</RichSelect.Option>
+    </RichSelect>
+  ),
+]
+
+export const OptionDisabled = Template.bind({})
+OptionDisabled.parameters = {
+  docs: {
+    storyDescription:
+      'Use `disabled` on `RichSelect.Option` component to disable a specific option.',
+  },
+}
+OptionDisabled.decorators = [
+  () => (
+    <RichSelect inputId="test" labelId="test-label" name="option-disabled">
+      <RichSelect.Option value="a">Option A</RichSelect.Option>
+      <RichSelect.Option value="b" disabled>
+        Option B
+      </RichSelect.Option>
+    </RichSelect>
+  ),
+]
+
+export const Animated = Template.bind({})
+Animated.parameters = {
+  docs: {
+    storyDescription: `You can easily add an animation to your RichSelect using \`animationOnChange\`, \`animation\` and \`animationDuration\` properties.
+The animation will be played when the value changes. Animation start when you select new value but also if you change the value of RichSelect with an external way (check example with button).
+
+### Available animations
+
+${Object.keys(animations)
+  .map(animation => `\`${animation}\``)
+  .join(' ')}`,
+  },
+}
+Animated.decorators = [
+  () => (
+    <Box mx={8} my={2}>
+      <ControlValue value="">
```

Comment:
```suggestion
      <ControlValue<string> value="">
```

To avoid eslint-disable below ?